react/components/AppVersionContext.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useContext, createContext } from 'react'
+import { useRuntime, withRuntimeContext } from 'vtex.render-runtime'
+import { compose, graphql } from 'react-apollo'
+
+import * as appMajorsQuery from '../graphql/getAppMajors.graphql'
+
+type Dispatch = (newMajorInfo: {
+  major: string
+  availableMajors: string[]
+}) => void
+interface State {
+  major: string
+  availableMajors: string[]
+}
+
+const AppVersionStateContext = createContext<State | undefined>(undefined)
+const AppVersionDispatchContext = createContext<Dispatch | undefined>(undefined)
+
+function AppVersionProvider({ children, appMajorsQuery }: any) {
+  const {
+    route: { params },
+  } = useRuntime()
+  const urlVersion = params.app.split('@')[1]
+  const hasVersion = !!urlVersion
+
+  const [versionInfo, setVersionInfo] = useState({
```

Comment:
Why not `useReducer`?

Reply:
I switched it to `useState` just for the sake of simplicity, as my state is not that complex and I don't see it changing much... I'm passing down to the `AppVersionDispatchContext.Provider` the `setState` function and I mutate it passing both state properties.

I do like the idea behind using a reducer function as you should me earlier so that your component's API scale better though. I can see where I would have problems updating a single state property at a time. Will update it :)